service/backend/pchain/indexer/parser.go

```diff
@@ -204,146 +175,55 @@ func (p *parser) ParseBlockWithHash(ctx context.Context, hash string) (*ParsedBl
-			errs.Add(p.initializeTx(tx))
-		}
-		parsedBlock.ParentID = castBlk.PrntID
-		parsedBlock.Txs = castBlk.Transactions
-		blockTimestamp = castBlk.Timestamp()
-	case *pBlocks.ApricotAbortBlock:
-		parsedBlock.ParentID = castBlk.PrntID
-		parsedBlock.Txs = []*txs.Tx{}
-	case *pBlocks.BanffAbortBlock:
-		parsedBlock.ParentID = castBlk.PrntID
-		parsedBlock.Txs = []*txs.Tx{}
-		blockTimestamp = castBlk.Timestamp()
-	case *pBlocks.BanffCommitBlock:
-		parsedBlock.ParentID = castBlk.PrntID
-		parsedBlock.Txs = []*txs.Tx{}
-		blockTimestamp = castBlk.Timestamp()
-	case *pBlocks.ApricotCommitBlock:
-		parsedBlock.ParentID = castBlk.PrntID
-		parsedBlock.Txs = []*txs.Tx{}
-	default:
-		errs.Add(fmt.Errorf("no handler exists for block type %T", castBlk))
-	}
-
-	// If no timestamp was found in a given block (pre-Banff) we fallback to proposer timestamp as used by Snowman++
-	// if available.
-	//
-	// For pre-Snowman++, proposer timestamp does not exist either. In that case, fallback to the genesis timestamp.
-	// This is needed as opposed to simply return 0 timestamp as Rosetta validation expects timestamps to be available
-	// after 1/1/2000.
-	if blockTimestamp.IsZero() {
-		if proposer.Timestamp > genesisTimestamp {
-			blockTimestamp = time.Unix(proposer.Timestamp, 0)
 		} else {
-			blockTimestamp = time.Unix(genesisTimestamp, 0)
+			proBlkData = Proposer{
+				ID:       proBlk.ID(),
+				ParentID: proBlk.ParentID(),
+			}
 		}
 	}
-	parsedBlock.Timestamp = blockTimestamp.UnixMilli()
-
-	return &parsedBlock, errs.Err
-}
-
-func getProposerFromBytes(bytes []byte) (Proposer, []byte, error) {
-	proposer, _, err := proposerBlk.Parse(bytes)
-	if err != nil || proposer == nil {
-		return Proposer{}, bytes, nil
-	}
 
-	switch castBlock := proposer.(type) {
-	case proposerBlk.SignedBlock:
-		return Proposer{
-			ID:           castBlock.ID(),
-			NodeID:       castBlock.Proposer(),
-			PChainHeight: castBlock.PChainHeight(),
-			Timestamp:    castBlock.Timestamp().Unix(),
-			ParentID:     castBlock.ParentID(),
-		}, castBlock.Block(), nil
-	case proposerBlk.Block:
-		return Proposer{}, castBlock.Block(), nil
-	default:
-		return Proposer{}, bytes, fmt.Errorf("no handler exists for proposer block type %T", castBlock)
-	}
-}
-
-// initializes tx to have tx identifier generated
-func (p *parser) initializeTx(tx *txs.Tx) error {
-	unsignedBytes, err := p.codec.Marshal(p.codecVersion, tx.Unsigned)
+	blk, err := pBlocks.Parse(p.codec, pChainBlkBytes)
 	if err != nil {
-		return err
+		return nil, fmt.Errorf("unmarshaling block bytes errored with %w", err)
 	}
 
-	signedBytes, err := p.codec.Marshal(p.codecVersion, tx)
-	if err != nil {
-		return err
+	txes := blk.Txs()
+	if txes == nil {
+		txes = []*txs.Tx{}
 	}
-
-	tx.Initialize(unsignedBytes, signedBytes)
-
-	return nil
+	return &ParsedBlock{
+		BlockID:   blk.ID(),
+		BlockType: fmt.Sprintf("%T", blk),
+		ParentID:  blk.Parent(),
+		Timestamp: container.Timestamp,
```

Comment:
We need this timestamp to be in milliseconds per https://www.rosetta-api.org/docs/models/NetworkStatusResponse.html.

Reply:
Fixed